sdks/python/apache_beam/runners/worker/data_plane_test.py

```diff
@@ -108,16 +106,28 @@ def send(instruction_id, transform_id, data):
         ])
 
     # Multiple interleaved writes to multiple instructions.
-    send('1', transform_1, b'abc')
-    send('2', transform_1, b'def')
+    stream11 = from_channel.output_stream('1', transform_1)
+    stream11.write(b'abc')
+    stream21 = from_channel.output_stream('2', transform_1)
+    stream21.write(b'def')
+    if not time_based_flush:
+      stream11.close()
     self.assertEqual(
         list(
             itertools.islice(to_channel.input_elements('1', [transform_1]), 1)),
         [
             beam_fn_api_pb2.Elements.Data(
                 instruction_id='1', transform_id=transform_1, data=b'abc')
         ])
-    send('2', transform_2, b'ghi')
+    if time_based_flush:
```

Comment:
Why do we need to wait for the flush, shouldn't the earlier `stream21.write(b'def')` provide the correct ordering?

Reply:
Write does not provide ordering guarantees in this case.

Elements are stored in a [queue](https://github.com/apache/beam/blob/7b3d4251d244c10545fb37f1d93ebcad84a98681/sdks/python/apache_beam/runners/worker/data_plane.py#L371) before being sent, to enable batching. Elements aren't added to that queue until the [flush callback](https://github.com/apache/beam/blob/7b3d4251d244c10545fb37f1d93ebcad84a98681/sdks/python/apache_beam/runners/worker/data_plane.py#L493) is invoked. Because the flush callback is [invoked periodically](https://github.com/apache/beam/blob/7b3d4251d244c10545fb37f1d93ebcad84a98681/sdks/python/apache_beam/runners/worker/data_plane.py#L182) starting from when a stream is constructed, there is no guarantee that one stream's callback is called before the other.

Reply:
Please add details as comment.

Reply:
I would have expected that `from_channel.output_stream` would share the same queue of elements when dealing with the same instruction id instead of two different queues with separate flushing characteristics.

Reply:
> Please add details as comment.

Done.